### PR TITLE
Add support for processing RequiredAttribute on OpenApi schemas

### LINKED ISSUE #56212: Properties with the `[Required]` attribute should be required in the generated OpenAPI document
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

A property of a class or record that has the `[Required]` attribute is currently not a required property of the schema produced by the .NET 9 OpenAPI document generation feature.

### Expected Behavior

A property with the `[Required]` attribute should be a required property of the OpenAPI schema.

### Steps To Reproduce

Not minimal, but there is a recreate in the "Schemas" section of my [aspnet-openapi-examples](https://github.com/mikekistler/aspnet-openapi-examples) project.

### Exceptions (if any)

_No response_

### .NET Version

9.0-preview4

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/dotnet/aspnetcore/issues/56212.

### CODE REVIEW

File: src/OpenApi/src/Services/OpenApiDocumentService.cs [hunk 1]
```diff
@@ -273,6 +273,15 @@ private async Task<OpenApiResponse> GetResponseAsync(ApiDescription apiDescripti
         return parameters;
     }
 
+    private static bool IsRequired(ApiParameterDescription parameter)
+    {
+        var hasRequiredAttribute = parameter.ParameterDescriptor is IParameterInfoParameterDescriptor parameterInfoDescriptor &&
+            parameterInfoDescriptor.ParameterInfo.GetCustomAttributes(inherit: true).Any(attr => attr is RequiredAttribute);
+        // Per the OpenAPI specification, parameters that are sourced from the path
+        // are always required, regardless of the requiredness status of the parameter.
+        return parameter.Source == BindingSource.Path || parameter.IsRequired || hasRequiredAttribute;
```
Comment (martincostello) [inline src/OpenApi/src/Services/OpenApiDocumentService.cs:282]: Is it worth refactoring to not run the code to find the attribute if the first two conditions are true?

Comment (captainsafia) [reply src/OpenApi/src/Services/OpenApiDocumentService.cs:282]: 🤷🏽‍♀️ I think the check is fairly cheap and I like that the current ordering is priority-based (e.g. makes it clear that a parameter being sourced from the path will always short-circuit the remaining checks).

File: src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.RequestBodySchemas.cs [hunk 2]
```diff
@@ -54,6 +55,56 @@ public async Task GetOpenApiRequestBody_GeneratesSchemaForPoco()
...
+        builder.MapPost("/required-properties", (RequiredTodo todo) => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            var operation = document.Paths["/required-properties"].Operations[OperationType.Post];
+            var requestBody = operation.RequestBody;
+            var content = Assert.Single(requestBody.Content);
+            var schema = content.Value.Schema;
+            Assert.Collection(schema.Required,
```
Comment (BrennanConroy) [inline src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.RequestBodySchemas.cs:102]: nit: `Assignee` should be checked for not required.

**Review by BrennanConroy — APPROVED** [review]

File: src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.RequestBodySchemas.cs (hunk 2 above)
Comment (captainsafia) [reply src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.RequestBodySchemas.cs:102]: The fact that it is not in the `schema.Required` list is indication enough but I can add a specific check to make it clearer.

**Comment by captainsafia** [comment]
@BrennanConroy I see you're on build-ops. Mind using your powers to merge this? The Helix failures are in the Mvc.Functional tests.

We should probably quarantine those too...

**Pull request merged**